hepstats/hypotests/calculators/frequentist_calculator.py

```diff
@@ -0,0 +1,430 @@
+import numpy as np
+from scipy.stats import norm
+
+from .basecalculator import BaseCalculator
+from ..fitutils.utils import pll, get_nevents
+from ..fitutils.sampling import base_sampler, base_sample
+from ..parameters import POI, POIarray
+
+
+class FrequentistCalculator(BaseCalculator):
+    """
+    Class for frequentist calculators.
+    """
+
+    def __init__(self, input, minimizer, ntoysnull=100, ntoysalt=100, sampler=base_sampler, sample=base_sample):
+        """Frequentist calculator class.
+
+            Args:
+                input : loss or fit result
+                minimizer : minimizer to use to find the minimum of the loss function
+                ntoysnull (int, default=100): number of toys to generate for the null hypothesis
+                ntoysalt (int, default=100): number of toys to generate for the alternative hypothesis
+                sampler : function used to create sampler with models, number of events and
+                    floating parameters in the sample Default is `hepstats.fitutils.sampling.base_sampler`.
+                sample : function used to get samples from the sampler.
+                    Default is `hepstats.fitutils.sampling.base_sample`.
+
+            Example with `zfit`:
+                >>> import zfit
+                >>> from zfit.core.loss import UnbinnedNLL
+                >>> from zfit.minimize import MinuitMinimizer
+
+                >>> obs = zfit.Space('x', limits=(0.1, 2.0))
+                >>> data = zfit.data.Data.from_numpy(obs=obs, array=np.random.normal(1.2, 0.1, 10000))
+                >>> mean = zfit.Parameter("mu", 1.2)
+                >>> sigma = zfit.Parameter("sigma", 0.1)
+                >>> model = zfit.pdf.Gauss(obs=obs, mu=mean, sigma=sigma)
+                >>> loss = UnbinnedNLL(model=[model], data=[data], fit_range=[obs])
+
+                >>> calc = FrequentistCalculator(input=loss, minimizer=MinuitMinimizer(), ntoysnull=1000, ntoysalt=1000)
+        """
+
+        super(FrequentistCalculator, self).__init__(input, minimizer)
+
+        self._toysnull = {}
+        self._toysalt = {}
+        self._ntoysnull = ntoysnull
+        self._ntoysalt = ntoysalt
+        self._sampler = sampler
+        self._sample = sample
+        self._toys_loss = {}
+        self._printlevel = 1
```

Comment:
This parameter is not settable and I can't remember where it comes from. Independently of whether it should be part of the arguments, maybe use an enum for possible values?

Reply:
Ok let's forget about this parameter for now x).

Reply:
But no need to remove it. My suggestion is simply to make this nicer in the medium term.

Reply:
BTW, click on the resolve buttons as you deal with things ...

Reply:
I mean it was used only at one place. We can think later about the verbosity of the calculator.